src/contracts/pods/EigenPod.sol

```diff
@@ -488,274 +451,182 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-         * as a way to "withdraw the same funds twice" without providing adequate proof.
-         * Note that this check is not made using the oracleTimestamp as in the `verifyWithdrawalCredentials` proof; instead this proof
-         * proof is made for the timestamp of the withdrawal, which may be within SLOTS_PER_HISTORICAL_ROOT slots of the oracleTimestamp.
-         * This difference in modifier usage is OK, since it is still not possible to `verifyAndProcessWithdrawal` against a slot that occurred
-         * *prior* to the proof provided in the `verifyWithdrawalCredentials` function.
-         */
-        proofIsForValidTimestamp(withdrawalProof.getWithdrawalTimestamp())
-        returns (VerifiedWithdrawal memory)
-    {
-        uint64 withdrawalTimestamp = withdrawalProof.getWithdrawalTimestamp();
-        bytes32 validatorPubkeyHash = validatorFields.getPubkeyHash();
+        _validatorPubkeyHashToInfo[proof.pubkeyHash] = validatorInfo;
+        emit ValidatorCheckpointed(checkpointTimestamp, validatorIndex);
+
+        return (balanceDeltaGwei, exitedBalanceGwei);
+    }
 
-        /**
-         * Withdrawal processing should only be performed for "ACTIVE" or "WITHDRAWN" validators.
-         * (WITHDRAWN is allowed because technically you can deposit to a validator even after it exits)
-         */
+    /**
+     * @dev Initiate a checkpoint proof by snapshotting both the pod's ETH balance and the
+     * current block's parent block root. After providing a checkpoint proof for each of the
+     * pod's ACTIVE validators, the pod's ETH balance is awarded shares and can be withdrawn.
+     * @dev ACTIVE validators are validators with verified withdrawal credentials (See
+     * `verifyWithdrawalCredentials` for details)
+     * @dev If the pod does not have any ACTIVE validators, the checkpoint is automatically
+     * finalized.
+     * @dev Once started, a checkpoint MUST be completed! It is not possible to start a
+     * checkpoint if the existing one is incomplete.
+     * @param revertIfNoBalance If the available ETH balance for checkpointing is 0 and this is
+     * true, this method will revert
+     */
+    function _startCheckpoint(bool revertIfNoBalance) internal {
         require(
-            _validatorPubkeyHashToInfo[validatorPubkeyHash].status != VALIDATOR_STATUS.INACTIVE,
-            "EigenPod._verifyAndProcessWithdrawal: Validator never proven to have withdrawal credentials pointed to this contract"
+            currentCheckpointTimestamp == 0, 
+            "EigenPod._startCheckpoint: must finish previous checkpoint before starting another"
         );
 
-        // Ensure we don't process the same withdrawal twice
         require(
-            !provenWithdrawal[validatorPubkeyHash][withdrawalTimestamp],
-            "EigenPod._verifyAndProcessWithdrawal: withdrawal has already been proven for this timestamp"
+            lastCheckpointTimestamp != uint64(block.timestamp),
```

Comment:
could we add a comment about the case this addresses?

Reply:
[4a923d0](https://github.com/Layr-Labs/eigenlayer-contracts/pull/515/commits/4a923d09fe9dbbf4c175502d410feef5d07e5b8d)